cmd/utils/flags.go

```diff
@@ -2158,12 +2166,15 @@ func MakeChain(ctx *cli.Context, stack *node.Node, readonly bool) (*core.BlockCh
 		cache.TrieDirtyLimit = ctx.Int(CacheFlag.Name) * ctx.Int(CacheGCFlag.Name) / 100
 	}
 	vmcfg := vm.Config{EnablePreimageRecording: ctx.Bool(VMEnableDebugFlag.Name)}
-
+	if ctx.IsSet(VMTraceFlag.Name) {
+		vmcfg.Tracer = tracers.NewPrinter()
```

Comment:
How did you envision this being customized to be a custom tracer code? Is there is still the JavaScript tracer that could be hooked here?

Would you guys be open to consider some kind of WASM tracer?

Reply:
> We have support for native Go tracers. I think that's what you should be using. When you use that your fork of geth should become a simple file-drop in the eth/tracers/native no other change needed.

Yeah that's exactly what I've started, a single file patch. 

 > I'll be adding logic to enable a choice of the native tracers. My first impression is we should leave out JS tracing for this feature. They would slow down block processing IMO.
 
 I was curious to see what you had in mind.

Reply:
Resolved

Reply:
I updated the way tracer is initialized. Now you can drop your tracer file in `eth/tracers/live` and add:

```go
func init() {
   register('name', constructor)
}
```

And then pass `--vmtrace name` which will activate the given live tracer. I suppose we can add in config object later if we want.